project/src/components/map/map.tsx

```diff
@@ -1,12 +1,49 @@
+import 'leaflet/dist/leaflet.css';
+import useMap from '../../hooks/useMap';
 import cn from 'classnames';
+import {useEffect, useRef} from 'react';
+import {Icon, Marker, LayerGroup} from 'leaflet';
+import {Offer, Location} from '../../types/offers-type';
+import {URL_MARKER_DEFAULT} from '../../constants';
 
 type MapProps = {
   className: string;
+  offers: Offer[];
+  city: Location;
 }
 
-function Map({className}: MapProps): JSX.Element{
+const defaultCustomIcon = new Icon({
+  iconUrl: URL_MARKER_DEFAULT,
+  iconSize: [30, 40],
+  iconAnchor: [15, 40]
+});
+
+function Map({className, offers, city}: MapProps): JSX.Element{
+  const mapRef = useRef(null);
+  const map = useMap(mapRef, city);
+
+  const markerGroup = new LayerGroup();
+
+  useEffect(() => {
+    if (map) {
+      offers.forEach((offer) => {
+        const marker = new Marker({
+          lat: offer.location.latitude,
+          lng: offer.location.longitude
+        });
+
+        marker
+          .setIcon(defaultCustomIcon)
+          .addTo(map);
+
+        markerGroup.addLayer(marker);
+      });
+    }
+    return () => {markerGroup.clearLayers();};
```

Comment:
Лайк за очистку только фигурки можно не ставить
```suggestion
    return () => markerGroup.clearLayers();
```